nucypher/blockchain/eth/actors.py

```diff
@@ -292,13 +297,22 @@ def connect_condition_providers(
                     raise self.ActorError(
                         f"Condition blockchain endpoint {uri} is not on chain {chain_id}"
                     )
-                providers[int(chain_id)].add(provider)
+                providers[int(chain_id)].append(provider)
+
+        # Ingest default RPC providers for each chain
+        for chain_id in self.domain.condition_chain_ids:
+            default_endpoints = get_healthy_default_rpc_endpoints(chain_id)
+            for uri in default_endpoints:
+                provider = self._make_condition_provider(uri)
+                providers[chain_id].append(provider)
+
+        self.log.info(f"Connected to {len(providers)} default RPC endpoints")
 
         humanized_chain_ids = ", ".join(
             _CONDITION_CHAINS[chain_id] for chain_id in providers
         )
         self.log.info(
-            f"Connected to {len(providers)} blockchains for condition checking: {humanized_chain_ids}"
+            f"Connected to {len(providers)} total endpoints for condition checking: {humanized_chain_ids}"
```

Comment:
`len(providers)` is really the number of chain ids, instead of the number of providers/endpoints - there are multiple endpoints per chain. 

You would have to sum the length of each value in the dictionary to get the total endpoints. Since you are going through the endpoints in the `for` loops above, you can keep a running count as you go, perhaps...?

Reply:
Taking these points into consideration, resolved it in a slightly different way.